src/plugins/vis_type_timeseries/server/routes/vis.ts

```diff
@@ -18,36 +18,19 @@
  */
 
 import { IRouter, KibanaRequest } from 'kibana/server';
-import { schema } from '@kbn/config-schema';
 import { getVisData, GetVisDataOptions } from '../lib/get_vis_data';
 import { visPayloadSchema } from './post_vis_schema';
-import { Framework, ValidationTelemetryServiceSetup } from '../index';
+import { Framework } from '../index';
 
-const escapeHatch = schema.object({}, { unknowns: 'allow' });
-
-export const visDataRoutes = (
-  router: IRouter,
-  framework: Framework,
-  { logFailedValidation }: ValidationTelemetryServiceSetup
-) => {
+export const visDataRoutes = (router: IRouter, framework: Framework) => {
   router.post(
     {
       path: '/api/metrics/vis/data',
       validate: {
-        body: escapeHatch,
+        body: visPayloadSchema,
       },
     },
     async (requestContext, request, response) => {
-      const { error: validationError } = visPayloadSchema.validate(request.body);
```

Comment:
Thank you for removing it. I like your idea of using official way for validating! 👍

Reply:
@DziyanaDzeraviankina Unfortunately we can't remove this part just yet. The telemetry is reporting that there are still some problems left in the validation which means working tsvb visualizations would start to fail when we are rolling this out now. The plan is to let the "soft validation" currently in place running for at least another minor.

The new kbn/schema should be used in the same way the joi schema is used right now.